### PR TITLE
Task 672, 750-754, 861-868: NumerSense Task 

### PR DESCRIPTION
**Pull request by tanay2001**
added the following  [task]( https://arxiv.org/pdf/2005.00683.pdf)

### CODE REVIEW

**Comment by swarooprm** [comment]
@tanay2001, thank you for creating tasks. However, I suggest you to take another dataset, as someone else has already created tasks from the Numersense dataset.

Here are several similar datasets which you can work on:

- https://github.com/CogComp/arithmetic (contains multiple datasets)
- https://github.com/deepmind/AQuA
- https://github.com/arkilpatel/SVAMP

**Comment by swarooprm** [comment]
@tanay2001, regarding this task: Definition, Positive and Negative examples seems to be from some other task, only instance field is from Numersense. Can you fix this so that we can review this task?

Since you have already created this task, we can review and approve this before you move to other datasets.

Also, make sure to rename your task to task672, as task numbers till 671 are already taken.

**Comment by tanay2001** [comment]
> @tanay2001, regarding this task: Definition, Positive and Negative examples seems to be from some other task, only instance field is from Numersense. Can you fix this so that we can review this task?
> 
> Since you have already created this task, we can review and approve this before you move to other datasets.
> 
> Also, make sure to rename your task to task672, as task numbers till 671 are already taken.

I missed that , have made the appropriate changes , I'll add the other datasets also in a while

**Comment by tanay2001** [comment]
> @tanay2001, thank you for creating tasks. However, I suggest you to take another dataset, as someone else has already created tasks from the Numersense dataset.
> 
> Here are several similar datasets which you can work on:
> 
> * https://github.com/CogComp/arithmetic (contains multiple datasets)
> * https://github.com/deepmind/AQuA
> * https://github.com/arkilpatel/SVAMP

Hi @swarooprm 
I have added the tasks you listed <br>

task 739 -->[https://aclanthology.org/N16-1136.pdf](https://aclanthology.org/N16-1136.pdf)
task 740 --> [https://aclanthology.org/2020.acl-main.92.pdf](https://aclanthology.org/2020.acl-main.92.pdf)
task 741 ---> AQuA
task 742-745 ---> SVAMP

**Comment by swarooprm** [comment]
Task672 looks good, some comments:
- The answer is wrong in 3rd positive example.
- There is some noise in the dataset e.g. "input": "Some plant varieties can grow up to <mask> feet tall.",
            "output": [
                "nine"
            ]
Clearly, nine is not the only answer and there can be other correct answers.
But I understand the noise is coming from the dataset and nothing much can be done here.

Will review other tasks shortly.

**Comment by swarooprm** [comment]
I have renamed tasks from task750-756 as task numbers before that were taken in previous PRs.

**Comment by swarooprm** [comment]
Tasks 750-756 look good. I like how you divided SVAMP into 4 different tasks, I was wondering if you can do something similar for other tasks e.g. the MAWPS corpus task consists of multiple datasets and each dataset can be a different task.
Here are the potential tasks:
addsub (addition, subtraction problems)
multiarith (multiplication, division)
singleop (single operation
singleq (single equation)
simuleq (simultaneous equations)

Similarly, in the case of AQUA and the other dataset, see if you can create multiple tasks.

Note that, by decomposing datasets into multiple simple tasks, we want to make the task relatively easy for models which are otherwise very difficult for models to solve.

Another general comment:
I see that you have mostly short and generic definitions. 
Make it more specific to the task e.g. in the case of the 'addsub' dataset you can append a statment in the definition something like this... "You need to perform addition/subtraction of numbers embedded in natural language in order to solve this task." Think about how will you describe a task to your friend, you will probably give some hints regarding the task; this will exactly be your definition.

**Comment by tanay2001** [comment]
> Tasks 750-756 look good. I like how you divided SVAMP into 4 different tasks, I was wondering if you can do something similar for other tasks e.g. the MAWPS corpus task consists of multiple datasets and each dataset can be a different task.
> Here are the potential tasks:
> addsub (addition, subtraction problems)
> multiarith (multiplication, division)
> singleop (single operation
> singleq (single equation)
> simuleq (simultaneous equations)
> 
> Similarly, in the case of AQUA and the other dataset, see if you can create multiple tasks.
> 
> Note that, by decomposing datasets into multiple simple tasks, we want to make the task relatively easy for models which are otherwise very difficult for models to solve.
> 
> Another general comment:
> I see that you have mostly short and generic definitions.
> Make it more specific to the task e.g. in the case of the 'addsub' dataset you can append a statment in the definition something like this... "You need to perform addition/subtraction of numbers embedded in natural language in order to solve this task." Think about how will you describe a task to your friend, you will probably give some hints regarding the task; this will exactly be your definition.

Hi @swarooprm 
I have incorporated the changes please review again

**Comment by swarooprm** [comment]
> > Tasks 750-756 look good. I like how you divided SVAMP into 4 different tasks, I was wondering if you can do something similar for other tasks e.g. the MAWPS corpus task consists of multiple datasets and each dataset can be a different task.
> > Here are the potential tasks:
> > addsub (addition, subtraction problems)
> > multiarith (multiplication, division)
> > singleop (single operation
> > singleq (single equation)
> > simuleq (simultaneous equations)
> > Similarly, in the case of AQUA and the other dataset, see if you can create multiple tasks.
> > Note that, by decomposing datasets into multiple simple tasks, we want to make the task relatively easy for models which are otherwise very difficult for models to solve.
> > Another general comment:
> > I see that you have mostly short and generic definitions.
> > Make it more specific to the task e.g. in the case of the 'addsub' dataset you can append a statment in the definition something like this... "You need to perform addition/subtraction of numbers embedded in natural language in order to solve this task." Think about how will you describe a task to your friend, you will probably give some hints regarding the task; this will exactly be your definition.
> 
> Hi @swarooprm I have incorporated the changes please review again

Thanks for incorporating feedback.
- Things have improved, however in some places the definition still looks generic like "answer the following question by choosing an option". Recent studies have shown that such generic definitions do not help models like GPT3. So please add a definition that is specific to the task e.g. you know that answering questions require some sort of mathematical operation. Often you also have knowledge about the type of operation (e.g. addition, subtraction) in some cases. I also appreciate that in some of your tasks the specific definition is already added.
- In many places, instances contain an additional .0 which is redundant. Examples are good and they don't contain decimal values. Please fix your instances by removing decimals wherever necessary.

Once these 2 issues get fixed, this PR is ready to get accepted @tanay2001

**Comment by tanay2001** [comment]
Some places removing the decimals has created an issue, I will fix this in the next commit and will get back

**Comment by tanay2001** [comment]
> > > Tasks 750-756 look good. I like how you divided SVAMP into 4 different tasks, I was wondering if you can do something similar for other tasks e.g. the MAWPS corpus task consists of multiple datasets and each dataset can be a different task.
> > > Here are the potential tasks:
> > > addsub (addition, subtraction problems)
> > > multiarith (multiplication, division)
> > > singleop (single operation
> > > singleq (single equation)
> > > simuleq (simultaneous equations)
> > > Similarly, in the case of AQUA and the other dataset, see if you can create multiple tasks.
> > > Note that, by decomposing datasets into multiple simple tasks, we want to make the task relatively easy for models which are otherwise very difficult for models to solve.
> > > Another general comment:
> > > I see that you have mostly short and generic definitions.
> > > Make it more specific to the task e.g. in the case of the 'addsub' dataset you can append a statment in the definition something like this... "You need to perform addition/subtraction of numbers embedded in natural language in order to solve this task." Think about how will you describe a task to your friend, you will probably give some hints regarding the task; this will exactly be your definition.
> > 
> > 
> > Hi @swarooprm I have incorporated the changes please review again
> 
> Thanks for incorporating feedback.
> 
> * Things have improved, however in some places the definition still looks generic like "answer the following question by choosing an option". Recent studies have shown that such generic definitions do not help models like GPT3. So please add a definition that is specific to the task e.g. you know that answering questions require some sort of mathematical operation. Often you also have knowledge about the type of operation (e.g. addition, subtraction) in some cases. I also appreciate that in some of your tasks the specific definition is already added.
> * In many places, instances contain an additional .0 which is redundant. Examples are good and they don't contain decimal values. Please fix your instances by removing decimals wherever necessary.
> 
> Once these 2 issues get fixed, this PR is ready to get accepted @tanay2001
@swarooprm 
I have addressed both the issues, in the latter one I made a mistake by just replacing all  ".0 " with  "" as a result, several cases of 2 decimal values were getting changed to a wrong value, I reverted the commit and did the substitution wisely so no issues exist in the latest version, hope its all good to go now

**Review by swarooprm — APPROVED** [review]
LGTM.
I assume labels are balanced in classification tasks, if not please correct in your next PR.

File: tasks/README.md [hunk 1]
```diff
@@ -557,6 +557,20 @@ Name | Summary | Category
...
+`task861_asdiv_addsub_question_answering` | Given a mathematical question , find the most suitable numerical answer| Answer Generation
+`task862_asdiv_multidiv_question_answering` |Given a mathematical question , find the most suitable numerical answer| Answer Generation
+`task863_asdiv_multiop_question_answering` |Given a mathematical question , find the most suitable numerical answer| Answer Generation
+`task864_asdiv_singleop_question_answering` |Given a mathematical question , find the most suitable numerical answer| Answer Generation
+`task865_mawps_addsub_question_answering` |Given a mathematical question , find the most suitable numerical answer|Answer Generation
+`task866_mawps_multidiv_question_answering` |Given a mathematical question , find the most suitable numerical answer| Answer Generation
+`task867_mawps_multiop_question_answering` |Given a mathematical question , find the most suitable numerical answer| Answer Generation
+`task868_mawps_singleop_question_answering` |Given a mathematical question , find the most suitable numerical answer|Answer Generation
```
Comment (danyaljj) [inline tasks/README.md:573]: @swarooprm when merging PRs, let's pay attention to the ordering of the tasks in the readme. Like in this case, these ideas are clearly misplaced.

File: tasks/README.md [hunk 2]
```diff
@@ -690,6 +704,7 @@ Name | Summary | Category
 `task641_esnli_classification` | Classification based on if two sentences agree, disasgree, or neutral. | Classification
 `task642_esnli_classification` | Classification based on if two statements agree/disagree or can't be determined. | Classification
 `task643_refresd_classification` | Classification based on if an english and french sentence are different or equivalent. | Classification
+`task644_refresd_translation` | Translation from english to french sentences. | Translation
```
Comment (danyaljj) [inline tasks/README.md:707]: @swarooprm another issue (duplicating the lines).

**Comment by swarooprm** [comment]
Yes, I agree. I see there are more such cases in the readme (task 399 before 377).
Yes, let's verify this while merging PR.

**Pull request merged**